compiler/modules/types/core.bal

```diff
@@ -815,13 +815,67 @@ function comparableNillableList(Context cx, SemType t1, SemType t2) returns bool
     }
     ComparableMemo memo = { semType1: t1, semType2: t2 };
     cx.comparableMemo.add(memo);
-    // SUBSET need to iterate members when tuples are supported
-    // following relies on the fact `listMemberType(cx, NIL, ()) = NEVER`
-    boolean result = comparable(cx, listMemberType(cx, t1, INT), listMemberType(cx, t2, INT));
+    boolean result;
+    var members1 = listAllMemberTypes(cx, t1);
+    var members2 = listAllMemberTypes(cx, t2);
+    // listAllMemberTypes return () if list type is not atomic
+    if members1 != () && members2 != () {
+        var mergedMembers = mergeListMemberTypes(members1, members2);
+        result = false;
+        foreach var [_, ty1, ty2] in mergedMembers {
+            result = comparable(cx, ty1, ty2);
+            memo.comparable = result;
+            if result == false {
+                return result;
+            }
+        }
+    }
+    else {
+        result = comparable(cx, listMemberType(cx, t1, INT), listMemberType(cx, t2, INT));
+    }
     memo.comparable = result;
     return result;
 }
 
+function mergeListMemberTypes([Range,SemType][] list1, [Range,SemType][] list2) returns [Range, SemType, SemType][] {
```

Comment:
Space after `Range,`

But perhaps we should define a type for `[Range,SemType][]` e.g. `MemberTypes`

Reply:
fixed